app/controllers/products_controller.rb

```diff
@@ -0,0 +1,51 @@
+class ProductsController < ApplicationController
+
+  def index
+    @categories = Category.order(name: :asc)
+    @merchants = User.joins(:products).uniq.sort
+
+    scope = Product.where(retired: false).order(name: :asc)
+
+    if params[:category].present?
```

Comment:
Consider whether you need `.present?`

Extension experiment:
Params is like a bucket. It doesn't have a box for category unless you had put it in there.
At this stage if `:category` is not present, would it be there at all?
would `if params[:category]` do the same thing?